### PR TITLE
Axes.inset_axes -  warning message remvoved

### LINKED ISSUE #27648: [Doc]: `Axes.inset_axes` is still experimental
### Documentation Link

https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.inset_axes.html#matplotlib.axes.Axes.inset_axes

### Problem

I just noticed that `Axes.inset_axes` is still marked as experimental.  It seems pretty well established to me, even recommended as a method for [placing colorbars](https://matplotlib.org/stable/users/explain/axes/colorbar_placement.html#using-inset-axes).

### Suggested improvement

Is it time to remove the "experimental" warning?

### PR DESCRIPTION
**Pull request by ShivamPathak99**
## Changed Axes.inset_axes status to stable #27648

Changed the status of Axes.inset_axes from experimental to stable in docs.
Created new file in next_whats_new for inset_axes.
Fixed the wrong fork

Closes #27648

## PR checklist

- [X] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [X] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [X] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [X] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [X] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/_api/__init__.py [hunk 1]
```diff
@@ -305,7 +305,6 @@ def my_func(*args, **kwargs):
                 [lambda old1, old2: locals(), lambda new: locals()],
                 *args, **kwargs)
             if "old1" in params:
-                warn_deprecated(...)
```
Comment (ksunden) [inline lib/matplotlib/_api/__init__.py:308]: Why was this removed? This is example code for certain deprecation procedures, while the `...` is in place of meaningful contextual inputs to `mpl._api.warn_deprecated` the fact that you want to warn in some cases is expected.

This is also unrelated to `inset_axes`

File: doc/users/next_whats_new/inset_axes.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+'Axes.inset_axes' is no longer experimental
+-------------------------
```
Comment (ksunden) [inline doc/users/next_whats_new/inset_axes.rst:2]:
```suggestion
'Axes.inset_axes' is no longer experimental
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

(Should match the length of the title and be tildes (this helps when combining the snippets into a single page as part of the release process.

File: doc/users/next_whats_new/inset_axes.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+Axes.inset_axes is now cosidered stable for use.
+
+
+~~~~~~~~~~~~~
```
Comment (ksunden) [inline doc/users/next_whats_new/inset_axes.rst:7]:
```suggestion
```

This is causing the doc build to fail

**Review by ksunden — CHANGES_REQUESTED** [review]
Mostly just doc formatting, but also one unrelated change is included

File: doc/users/next_whats_new/inset_axes.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+'Axes.inset_axes' is no longer experimental
+-------------------------------------------
```
Comment (ksunden) [inline doc/users/next_whats_new/inset_axes.rst:2]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

please use tildes

**Comment by ShivamPathak99** [comment]
Why 2 of the checks failed ?

**Comment by ksunden** [comment]
One is an unrelated memory problem we have been having (nothing for you to do on that)

The other is wanting the new line at the end of the release note to be taken out, I believe.

File: doc/users/next_whats_new/inset_axes.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
+``Axes.inset_axes`` is no longer experimental
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Axes.inset_axes is cosidered stable for use.
```
Comment (jklymak) [inline doc/users/next_whats_new/inset_axes.rst:4]:
```suggestion
Axes.inset_axes is considered stable for use.
```

Comment (ShivamPathak99) [reply doc/users/next_whats_new/inset_axes.rst]: Done

File: doc/users/next_whats_new/inset_axes.rst (hunk 2 above)
Comment (ShivamPathak99) [reply doc/users/next_whats_new/inset_axes.rst]: Done

**Comment by ShivamPathak99** [comment]
`codecov/project/tests ` is failing, but why ?

**Comment by tacaswell** [comment]
I'm unclear why the bot added "documention user guide" as this is moving part of the API from experimental to stable so it is closest to an API change. 

---


> Coverage data is based on HEAD 95b7f68
(10 uploads)
 compared to BASE 879bde7
(14 uploads)


more CI jobs contributed to the run on the base commit.

From looking at the indirect changes it looks like we are missing an upload from an osx runner and something with a pre-release of pytest8.

**Comment by ksunden** [comment]
> I'm unclear why the bot added "documention user guide"

The release note matches this file glob:

https://github.com/matplotlib/matplotlib/blob/59f9b3c707da6888f0053cf66a2e61b25a21bd3b/.github/labeler.yml#L104-L107

Perhaps that glob can be narrowed (especially for `next_whats_new`)

**Review by timhoffm — APPROVED** [review]

**Review by dstansby — APPROVED** [review]
